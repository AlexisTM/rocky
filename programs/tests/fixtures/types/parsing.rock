Say "parsing ints with cast"
Let X be "123"
Let Y be "456"
Let Z be X with Y
Shout Z (123456)
Cast X
Cast Y
Let Z be X with Y
Shout Z (579)

Say "parsing double with cast"
Let X be "123.45"
Cast X
Let Z be X with Y
Shout Z (579.45)

Say "parsing strings with radix cast"
Let X be "ff"
Cast X with 16
Shout X (170)
Cast "10101010" into Z with 2
Shout Z (170 - hex aa)

Say "creating chars with cast"
Let X be 65
Cast X
Shout X (A)

Say "Conversion types: in-place"
Let X be 65
Cast X
Shout X (A)

Say "Conversion types: into"
Let X be 65
Cast X into Z
Shout X (65)
Shout Z (A)

Say "Conversion types: with"
Let X be "ff"
Cast X with 16
Shout X (255)

Say "Conversion types: into-with"
Let X be "ff"
Cast X with 16 into Z
Shout X (ff)
Shout Z (255)

Say "Conversion complex cases"
Cast "1010" with (+) "0101" into Z with 2
Say Z (165)
Cast "10101010" into Z with 2
Say Z (170)
Cast "10101001" with 2 into Z
Say Z (169)
