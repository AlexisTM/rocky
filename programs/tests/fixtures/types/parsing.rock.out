parsing ints with cast
123456
579
parsing double with cast
579.45
parsing strings with radix cast
255
170
creating chars with cast
A
Conversion types: in-place
A
Conversion types: into
65
A
Conversion types: with
255
Conversion types: into-with
ff
255
Conversion complex cases
165
170
169
